src/Executor/Executor.php

```diff
@@ -173,10 +173,12 @@ public static function defaultFieldResolver($objectValue, $args, $contextValue,
         $fieldName = $info->fieldName;
         $property  = null;
 
-        if (is_array($objectValue) || $objectValue instanceof ArrayAccess) {
+        if (is_array($objectValue)) {
             if (isset($objectValue[$fieldName])) {
                 $property = $objectValue[$fieldName];
             }
+        } elseif ($objectValue instanceof ArrayAccess) {
```

Comment:
An explicit test case for this would be great.

Reply:
Sure, but I need some pointers. There's no unit test for `defaultFieldResolver()`. Any pointers to test code that I could use as a base?

Reply:
> Sure, but I need some pointers.

Here you go:

```c
int *foo;
char *bar;
```

*Sorry, couldn't resist.*

I am assuming there is a test class that covers this entire class.

Reply:
I just dug around https://github.com/webonyx/graphql-php/blob/master/tests/Executor/ExecutorTest.php a bit. Given this library is mostly a port of the JS reference implementation, it is possible there are no explicit tests for the default resolver, given that JS only has objects and regular property access.

Reply:
```
Segmentation fault
```

Reply:
Perhaps `testExecutesArbitraryCode` could have yet another element added to its already huge array, with a ArrayAccessor instance instead of a pure array. Something like `deep`, a `deepAccessor`. Since there is no coverage there were no previous tests checking this if branch. If you have a better idea give me some extra pointers. But make them a pointer to a reference to a pointer this time.
I could also write a basic unit test for it, but it'd be completely useless.

Reply:
A dedicated test method would be better, since the existing method should mirror the JS implementation. Something like `testDefaultResolverHandlesArrayAccess`?

Reply:
I added a comprehensive test case to reify the current behaviour: https://github.com/webonyx/graphql-php/pull/761